keysign/discover.py

```diff
@@ -42,17 +42,19 @@ def start(self):
         if self.bt_code and not self.stopped:
             # We try Bluetooth, if we have it
             log.info("Trying to connect to %s with Bluetooth", self.bt_code)
-            self.bt = BluetoothReceive(self.bt_port)
-            msg_tuple = yield self.bt.find_key(self.bt_code, self.mac)
-            key_data, success, message = msg_tuple
-            if key_data:
+            if BluetoothReceive is not None:
+                self.bt = BluetoothReceive(self.bt_port)
+                msg_tuple = yield self.bt.find_key(self.bt_code, self.mac)
+                key_data, success, message = msg_tuple
+                if key_data:
                 # If we found the key
-                returnValue((key_data, success, message))
-
-        key_data = None
-        success = False
-        message = ""
-        returnValue((key_data, success, message))
+                    returnValue((key_data, success, message))
+            
+    log.info("PyBluez is probably missing.")
```

Comment:
I realized that here on Github is not very clear which line I highlighted.
In my previous comment I was referring only to the `key_data = None`, because it makes sense to keep the log inside the else.
Otherwise we will get the log about the missing PyBluez also when we simply don't have a `bt_code`, that is wrong.
So simply place only the log in the `else` like it was previously  leaving the rest as it is now, and we are good :)

Reply:
I was wondering that. :D